core/model/src/main/java/org/eclipse/rdf4j/model/util/ModelBuilder.java

```diff
@@ -0,0 +1,357 @@
+	 */
+	public ModelBuilder add(IRI predicate, Object object) {
+		if (currentSubject == null) {
+			throw new ModelException("subject not set");
+		}
+		return add(currentSubject, predicate, object);
+	}
+
+	/**
+	 * Add an RDF statement with the predicate and object to the model, using the current subject and graph
+	 * (either named or default).
+	 * 
+	 * @param predicate
+	 *        the statement's predicate. This can be defined either as a prefixed name string (e.g.
+	 *        "ex:john"), or as a full IRI (e.g. "http://example.org/john"). If supplied as a prefixed name,
+	 *        the {@link ModelBuilder} will need to have a namespace mapping for the prefix.
+	 * @param object
+	 *        the statement's object. If the supplied object is a {@link BNode}, {@link IRI}, or
+	 *        {@link Literal}, the object is used directly. Otherwise this method creates a typed
+	 *        {@link Literal} out of the supplied object, mapping the runtime type of the object to the
+	 *        appropriate XML Schema type. If no mapping is available, the method creates a literal with the
+	 *        string representation of the supplied object as the value, and {@link XMLSchema#STRING} as the
+	 *        datatype. Recognized types are {@link Boolean} , {@link Byte}, {@link Double}, {@link Float},
+	 *        {@link Integer}, {@link Long}, {@link Short}, {@link XMLGregorianCalendar } , and {@link Date}.
+	 * @return this {@link ModelBuilder}
+	 * @throws ModelException
+	 *         if the current subject is not set using {@link #subject(Resource)} or {@link #subject(String)}.
+	 */
+	public ModelBuilder add(String predicate, Object object) {
+		return add(mapToIRI(predicate), object);
+	}
+
+	/**
+	 * Return the created {@link Model}
+	 * 
+	 * @return the {@link Model}
+	 */
+	public Model build() {
+		return model;
+	}
+
+	private IRI mapToIRI(String prefixedNameOrIRI) {
+		if (prefixedNameOrIRI.indexOf(':') < 0) {
+			throw new ModelException("invalid prefixed name or IRI: " + prefixedNameOrIRI);
+		}
+
+		final String prefix = prefixedNameOrIRI.substring(0, prefixedNameOrIRI.indexOf(':'));
+
+		final ValueFactory vf = SimpleValueFactory.getInstance();
+
+		for (Namespace ns : model.getNamespaces()) {
+			if (prefix.equals(ns.getPrefix())) {
+				return vf.createIRI(ns.getName(),
+						prefixedNameOrIRI.substring(prefixedNameOrIRI.indexOf(':') + 1));
+			}
+		}
+
+		// try mapping using some of the default / well-known namespaces
+		for (Namespace ns : getDefaultNamespaces()) {
+			if (prefix.equals(ns.getPrefix())) {
+				model.setNamespace(ns);
+				return vf.createIRI(ns.getName(),
+						prefixedNameOrIRI.substring(prefixedNameOrIRI.indexOf(':') + 1));
+			}
+		}
+		return vf.createIRI(prefixedNameOrIRI);
+	}
+
+	private Namespace[] getDefaultNamespaces() {
+		return new Namespace[] { RDF.NS, RDFS.NS, OWL.NS, XMLSchema.NS, DC.NS, FOAF.NS, SKOS.NS };
```

Comment:
Although there has been extensive discussion about the prefix for DCTERMS (we currently use dcterms but others prefer either dc or dct), it should be represented in the list here as there are many additions that don't map onto the original DC Elements 1.1 vocabulary.